apcd-cms/src/apps/registrations/views.py

```diff
@@ -21,13 +23,21 @@ class SubmissionFormView(View):
     def get(self, request):
         formatted_reg_data = []
         renew = False
-        if 'reg_id' in request.GET:
-            reg_id = request.GET.get('reg_id')
-            renew = True
-            registration_content = apcd_database.get_registrations(reg_id)[0]
-            registration_entities = apcd_database.get_registration_entities(reg_id)
-            registration_contacts = apcd_database.get_registration_contacts(reg_id)
-            formatted_reg_data = _set_registration(registration_content, registration_entities, registration_contacts)
+        reg_id = request.GET.get('reg_id', None)
+        if reg_id and (apcd_database.get_user_role(request.user.username) in ['APCD_ADMIN', 'SUBMITTER_ADMIN']):
```

Comment:
I tried putting in a hardcoded real user name ('thbrown' for example) here and on line 29 in lieu of the request.user to test. When I clicked the renewal action in the listing dropdown, the url passed me on to a blank form template. Great work! 

On the form page, the url has a query param of reg_id=1 even though the form is blank. Will this form on post take this query param at any point? Maybe the reg_id parameter should be cleared after this check in the registration view if it doesn't pass the requirements for pre-populating?

Reply:
Good point, I'll look into it!